### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'ingredients')

### ISSUE THREAD

**Issue by jrcichra** [Sleavely/satisfactory-savegame-prometheus-exporter#10]
Seeing this error which fails all the other scrapes:

```
satisfactory-0 savegame-prometheus-exporter TypeError: Cannot read properties of undefined (reading 'ingredients')
satisfactory-0 savegame-prometheus-exporter     at parser6 (/app/dist/express.js:51410:37)
satisfactory-0 savegame-prometheus-exporter     at extractMetrics (/app/dist/express.js:51491:7)
```

Will try to look into what object it is failing on but the stacktrace doesn't seem to be full.

**Comment by jrcichra** [comment]
If I remove the exception handling in the CLI I get a line number in the Javascript:
```
/app/dist # ./cli.js
/app/dist/cli.js:51355
    for (const ingredient of recipe.ingredients) {
                                    ^
```

**Comment by jrcichra** [comment]
I'm not sure what object is breaking the exporter but this will let it keep exporting: https://github.com/Sleavely/satisfactory-savegame-prometheus-exporter/pull/11

**Comment by Sleavely** [comment]
Nice detective work.

Aside from your supplied fix, my two takeaways are:
* The image should include sourcemaps to make debugging a bit easier
* The error handling overall could use an overhaul to be more graceful and log warnings rather than throwing fatal errors.

**Comment by Sleavely** [comment]
Closing as resolved since your fix was published in [v2.2.3](https://github.com/Sleavely/satisfactory-savegame-prometheus-exporter/releases/tag/v2.2.3)
